zipline/gens/tradesimulation.py

```diff
@@ -100,6 +101,16 @@ def transform(self, stream_in):
                 self.simulation_dt = date
                 self.on_dt_changed(date)
 
+                # removing expired futures
+                for sid in self.current_data.keys():
+                    try:
+                        if self.env.asset_finder.retrieve_asset(sid).end_date \
+                                < self.simulation_dt:
+                            self.current_data.__delitem__(sid)
+                    except (AttributeError, TypeError, ValueError,
```

Comment:
Two additional notes here:
1. `self.current_data.__delitem_(key)` is better written as `del self.current_data[key]`.
2. This will fail on Python3 because `keys()` is a view on the dictionary, and you're not allowed to mutate a dictionary in place while iterating over a view:
```
Python 3.4.3 (default, Oct 14 2015, 20:28:29)
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> x = {1:1, 2: 2, 3: 3}
>>> for k in x.keys():
...     del x[k]
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dictionary changed size during iteration
```
Compare to:
```
>>> x = {1:1, 2: 2, 3: 3}
>>> for k in x.keys():
...     del x[k]
...
>>> x
{}
```